pytools/graph.py

```diff
@@ -45,13 +46,37 @@
     Any type.
 """
 
-from typing import (TypeVar, Mapping, Iterable, List, Optional, Any, Callable,
-                    Set, MutableSet, Dict, Iterator, Tuple)
+from typing import (Collection,  TypeVar, Mapping, Iterable, List, Optional, Any,
+                    Callable, Set, MutableSet, Dict, Iterator, Tuple, FrozenSet)
 
 
 T = TypeVar("T")
 
 
+# {{{ reverse_graph
+
+def reverse_graph(graph: Mapping[T, Collection[T]]) -> Dict[T, FrozenSet[T]]:
+    """
+    Reverses a graph.
+
+    :param graph: A :class:`dict` representation of a directed graph, mapping each
+        node to other nodes to which it is connected by edges.
```

Comment:
Doesn't explicitly talk about the direction of the edge.

Reply:
I thought about that! I ended up deciding that there was no need to document this, as it's just the reverse of what it was before, no matter which way they were supposed to be pointing initially.

Reply:
Fair, sg!